hoomd/write/gsd_burst.py

```diff
@@ -0,0 +1,106 @@
+# Copyright (c) 2009-2023 The Regents of the University of Michigan.
+# Part of HOOMD-blue, released under the BSD 3-Clause License.
+
+"""Write GSD last :math:`N` frames at user direction."""
+
+from hoomd import _hoomd
+from hoomd.filter import All
+from hoomd.data.parameterdicts import ParameterDict
+from hoomd.write.gsd import GSD
+
+
+class Burst(GSD):
+    r"""Write the last :math:`N` stored frames in the GSD format.
+
+    When triggered, `Burst` stores up to the last :math:`N` frames in a buffer.
+    `Burst` writes the contents of the buffer to the file when `dump` is called.
+    When the the next frame would result in :math:`N + 1` frames being stored,
+    the oldest frame is removed and the new frame is added.
+
+    Args:
+        trigger (hoomd.trigger.trigger_like): Select the timesteps to store
+            in the buffer.
+        filename (str): File name to write.
+        filter (hoomd.filter.filter_like): Select the particles to write.
+            Defaults to `hoomd.filter.All`.
+        mode (str): The file open mode. Defaults to ``'ab'``.
+        dynamic (list[str]): Field names and/or field categores to save in
+            all frames. Defaults to ``['property']``.
+        logger (hoomd.logging.Logger): Provide log quantities to write. Defaults
+            to `None`.
+        max_burst_frames (int): The maximum number of frames to store before
+            between writes. -1 represents no limit. Defaults to -1.
+        write_at_start (bool): When ``True`` **and** the file does not exist or
```

Comment:
`write_at_start` seems like the wrong name for this. Consider using `write_initial_frame`.

Reply:
Decided offline that the current name is okay.